File: python/parse_json_without_keys.py
"""
This code fetch the values of specified keys from a nested JSON file recursively.
Input: JSON file
Output: List of values
"""

import json


def json_extract(json_object):
    """Recursively fetch values from nested JSON."""
    values_list = []

    def extract(obj):
        """Recursively search for values of key in JSON tree."""
        resource_id = ""
        text = ""
        # Provide the keys for which you want the values.
        if "resource-id" in obj:
            resource_id = obj["resource-id"]
        if "text" in obj:
            text = obj["text"]
        values_list.append({"resource-id": resource_id, "text": text})

        # Provide the key for which you want to get the values recursively.
        if "children" in obj:
            children = obj['children']
            for elements in children:
                extract(elements)

    extract(json_object)
    return values_list


if __name__ == '__main__':
    json_file_path = '../data/example.json'
    with open(json_file_path, 'r') as json_file:
        json_file_data = json.load(json_file)
        activity = json_file_data["activity"]
        root = activity["root"]
        values = json_extract(root)     # I am providing "root" as the JSON object. You can give "json_file_data" instead.
        print(values.__len__())
        print(values)

File: src/main/java/ParseJSONWithoutKeys.java
/*
This code fetch the values of specified keys from a nested JSON file recursively.
Input: JSON file
Output: Values
*/
import org.json.JSONArray;
import org.json.JSONObject;
import org.json.simple.parser.JSONParser;
import java.io.FileReader;

public class ParseJSONWithoutKeys {
    public static void getValues(JSONObject object)
    {
        // Provide the key for which you want the values.
        String text;
        if(object.has("text"))
        {
            text = (String) object.get("text");
            System.out.println(text);
        }
        // Provide the key for which you want to get the values recursively.
        if(object.has("children"))
        {
            JSONArray jsonarray = object.getJSONArray("children");
            for(int i = 0; i < jsonarray.length(); i++)
            {
                String jsonArrayString = jsonarray.get(i).toString();
                JSONObject innerJSOn = new JSONObject(jsonArrayString);
                getValues(innerJSOn);
            }
        }
    }

    public static void main(String [] args)
    {
        String jsonFilePath = "data/example.json";
        JSONParser jsonParser = new JSONParser();
        JSONObject inputJSONObject;

        try (FileReader reader = new FileReader(jsonFilePath))
        {
            String obj = jsonParser.parse(reader).toString();
            inputJSONObject = new JSONObject(obj);
            JSONObject activity = inputJSONObject.getJSONObject("activity");
            JSONObject root = activity.getJSONObject("root");

            getValues(root);

        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }
}